src/screens/TokenDetail.js

```diff
@@ -21,85 +21,66 @@ import { TX_COUNT } from '../constants';
-   * paramUID {string} UID of token in the URL
    * transactions {Array} Array of transactions for the token
    */
   state = {
     token: null,
-    successMessage: '',
     errorMessage: '',
-    totalSupply: 0,
-    canMint: null,
-    canMelt: null,
-    paramUID: '',
     transactions: [],
   };
 
   componentDidMount() {
     const { match: { params } } = this.props;
 
-    this.setState({ paramUID: params.tokenUID }, () => {
-      this.updateTokenInfo();
-    });
+    this.setTokenId(params.tokenUID);
+  }
+
+  setTokenId(id) {
+    this.updateTokenInfo(id);
+    this.updateTokenMetadata(id);
   }
 
   /**
    * Upadte token info getting data from the full node (can mint, can melt, total supply)
    */
-  updateTokenInfo = () => {
-    hathorLib.walletApi.getGeneralTokenInfo(this.state.paramUID, (response) => {
+  updateTokenInfo = (id) => {
+    hathorLib.walletApi.getGeneralTokenInfo(id, (response) => {
       if (response.success) {
-        this.setState({
-          token: {uid: this.state.paramUID, name: response.name, symbol: response.symbol },
-          totalSupply: response.total,
-          canMint: response.mint.length > 0,
-          canMelt: response.melt.length > 0,
-          transactionsCount: response.transactions_count,
+        this.setState((oldState) => {
+          return {
+            token: {
+              ...oldState.token,
+              uid: id,
+              name: response.name,
+              symbol: response.symbol,
+              totalSupply: response.total,
+              canMint: response.mint.length > 0,
+              canMelt: response.melt.length > 0,
+              transactionsCount: response.transactions_count,
+            },
+          }
         });
       } else {
         this.setState({ errorMessage: response.message });
       }
     });
   }
 
-  /**
-   * Called when user clicks to download the qrcode
-   * Add the href from the qrcode canvas
-   *
-   * @param {Object} e Event emitted by the link clicked
-   */
-  downloadQrCode = (e) => {
-    e.currentTarget.href = document.getElementsByTagName('canvas')[0].toDataURL();
-  }
-
-  /**
-   * Show alert success message
-   *
-   * @param {string} message Success message
-   */
-  showSuccess = (message) => {
-    this.setState({ successMessage: message }, () => {
-      this.refs.alertSuccess.show(3000);
-    })
-  }
-
-  /**
-   * Method called on copy to clipboard success  
-   * Show alert success message
-   *
-   * @param {string} text Text copied to clipboard
-   * @param {*} result Null in case of error
-   */
-  copied = (text, result) => {
-    if (result) {
-      // If copied with success
-      this.showSuccess('Configuration string copied to clipboard!');
-    }
+  updateTokenMetadata = (id) => {
+    tokenApi.getMetadata(id).then((data) => {
+      if (data) {
+        this.setState((oldState) => {
+          return {
+            token: {
+              ...oldState.token,
+              uid: id,
```

Comment:
Why do you set the `uid` again?

Reply:
Maybe it's because `updateTokenMetadata` and `updateTokenInfo` are async calls, then either of them could finish first. And if he calls `setState` without a `this.state.token.uid` it will break the component jsx, is that right?

I like having both API calls async to be honest, maybe we should set the token uid on state before the API calls.

Reply:
There's a bug on `Transactions` component that it is not properly reactive. If the `token` is set without `uid` it makes a resquest to retrieve the list with `undefined` and never rerender again.

As the request to metadata service can finish first than the tokendetails, we have to set the `uid` when setting the token.

Reply:
You can always substitute

```
setTokenId(id) {
  this.updateTokenInfo(id);
  this.updateTokenMetadata(id); 
}
```

with

```
setTokenId(id) {
  this.setState({ token: { uid: id } }, () => {
    this.updateTokenInfo(id);
    this.updateTokenMetadata(id); 
  });
}
```

Then you won't need to set the uid again in both responses. But I am also okay to keep like it's coded right now.